src/complex/DataStructure/INeighborList.hpp

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+#include "complex/DataStructure/IDataArray.hpp"
+
+namespace complex
+{
+/**
+ * @brief Non-templated base class for NeighborList class.
+ */
+class COMPLEX_EXPORT INeighborList : public IDataArray
```

Comment:
Do we want `INeighborList` to inherit from `IDataArray`?

Reply:
For the sake of complexity, that needs to be its own PR. I'm stashing my changes for now to create a new branch with all the required changes. ArraySelectionParameter needs to be split for the two "array" types, filters need to be updated, and various helper methods need their parameters changed.

This PR is about HDF5, so the changes should be kept to that topic even if it lays the groundwork for additional changes. The PR would get way too much bloat otherwise.